Source/Extensions/Blazorise.Components/Autocomplete.razor.cs

```diff
@@ -118,28 +104,14 @@ public override async Task SetParametersAsync( ParameterView parameters )
                 selectedText = null;
             }
 
-            if ( parameters.TryGetValue<IEnumerable<TValue>>( nameof( SelectedValues ), out var paramSelectedValues ) && !SequenceEqual( selectedValuesParam, paramSelectedValues ) )
+            if ( parameters.TryGetValue<IEnumerable<TValue>>( nameof( SelectedValues ), out var paramSelectedValues ) && selectedValuesParam != paramSelectedValues )
             {
-                if ( selectedValues != paramSelectedValues )
-                {
-                    selectedValues.Clear();
-                    if (paramSelectedValues != null)
-                    {
-                        selectedValues.AddRange( paramSelectedValues );
-                    }
-                }
+                selectedValues = null;
             }
 
-            if ( parameters.TryGetValue<IEnumerable<string>>( nameof( SelectedTexts ), out var paramSelectedTexts ) && !SequenceEqual( selectedTextsParam, paramSelectedTexts ) )
+            if ( parameters.TryGetValue<IEnumerable<string>>( nameof( SelectedTexts ), out var paramSelectedTexts ) && selectedTextsParam != paramSelectedTexts )
             {
-                if ( selectedTexts != paramSelectedTexts )
-                {
-                    selectedTexts.Clear();
-                    if ( paramSelectedTexts != null )
-                    {
-                        selectedTexts.AddRange( paramSelectedTexts );
-                    }
-                }
```

Comment:
So I replaced it with what I've been calling a three variable pattern. You have the property value, old parameter value and the new parameter value. I found it's useful to imagine a table like this
<img width="183" alt="image" src="https://user-images.githubusercontent.com/22550674/182033742-24737175-2d4a-4fcb-834c-3ee39006cfcd.png">
and consider how binding will work in each of the cases. 
So for instance in Case A:

1. After initialization: property = null, old param = null
2. Afer you select an item: property = new List() { item }, old param = null
3. GetProperty() => property ?? old param;

For case C

1. After initialization: property = null, old param = null
2. After you select an item:
2.1 property = new List() { item }
2.2 fire property changed - bound variable updated
2.3 SetParametersAsync - property = null, old param = new param
3. GetProperty() => property ?? old param;

I think this is a much cleaner approach than what was there before with only two variables.

Reply:
I removed SequenceEqual since I don't need to compare values because I'm just taking the container reference with all the container contents

Reply:
> I removed SequenceEqual since I don't need to compare values because I'm just taking the container reference with all the container contents

So if for some reason the client provides a new reference for, let's imagine SelectedValues, with the same container elements, what happens?